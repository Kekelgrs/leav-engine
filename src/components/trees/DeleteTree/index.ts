import DeleteTree from './DeleteTree';
export default DeleteTree;
