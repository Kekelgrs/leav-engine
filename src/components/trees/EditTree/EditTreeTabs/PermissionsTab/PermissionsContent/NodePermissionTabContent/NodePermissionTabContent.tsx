import React from 'react';
import {useTranslation} from 'react-i18next';
import {Tab} from 'semantic-ui-react';
import useLang from '../../../../../../../hooks/useLang';
import {localizedLabel} from '../../../../../../../utils';
import {
    GET_TREE_BY_ID_trees_list,
    GET_TREE_BY_ID_trees_list_libraries,
    GET_TREE_BY_ID_trees_list_permissions_conf_permissionsConf_permissionTreeAttributes_TreeAttribute
} from '../../../../../../../_gqlTypes/GET_TREE_BY_ID';
import {
    AttributeType,
    PermissionsRelation,
    PermissionTypes,
    Treepermissions_confInput
} from '../../../../../../../_gqlTypes/globalTypes';
import DefinePermByUserGroupView from '../../../../../../permissions/DefinePermByUserGroupView';
import DefineTreePermissionsView from '../../../../../../permissions/DefineTreePermissionsView';
import PermissionsAttributesSelector from '../../../../../../permissions/PermissionsAttributesSelector';

interface INodePermissionTabContentProps {
    tree: GET_TREE_BY_ID_trees_list;
    library: GET_TREE_BY_ID_trees_list_libraries;
    onSubmitSettings: (library: string, conf: Treepermissions_confInput) => void;
    readonly: boolean;
}

function NodePermissionTabContent({
    readonly,
    library,
    tree,
    onSubmitSettings
}: INodePermissionTabContentProps): JSX.Element {
    const {lang} = useLang();
    const {t} = useTranslation();
    const _handleSubmit = (conf: Treepermissions_confInput) => onSubmitSettings(library.id, conf);

    const attributes = library.attributes ? library.attributes.filter(a => a.type === AttributeType.tree) : [];
    const treePermsConf = tree.permissions_conf?.filter(p => p.libraryId === library.id)?.[0]?.permissionsConf ?? null;

    const permissionsPanes = treePermsConf?.permissionTreeAttributes
        ? treePermsConf?.permissionTreeAttributes.map(a => ({
              key: 'a.id',
              menuItem: localizedLabel(a.label, lang),
              render: () => (
                  <Tab.Pane key="treePermissions" className="grow flex-col height100">
                      {
                          <DefineTreePermissionsView
                              key={a.id}
                              treeAttribute={
                                  a as GET_TREE_BY_ID_trees_list_permissions_conf_permissionsConf_permissionTreeAttributes_TreeAttribute
                              }
                              permissionType={PermissionTypes.tree_node}
                              applyTo={`${tree.id}/${library.id}`}
                              readOnly={readonly}
                          />
                      }
                  </Tab.Pane>
              )
          }))
        : [];

    permissionsPanes.unshift({
        key: 'libPermissions',
        menuItem: t('permissions.library_tab_name'),
        render: () => (
            <Tab.Pane key="libPermissions" className="grow flex-col height100">
                {
                    <DefinePermByUserGroupView
                        type={PermissionTypes.tree_library}
                        key="libPermissions"
                        applyTo={`${tree.id}/${library.id}`}
                        readOnly={readonly}
                    />
                }
            </Tab.Pane>
        )
    });

    return (
        <>
            <PermissionsAttributesSelector
                attributes={attributes}
                readonly={readonly}
                permissionsConf={{
                    relation: treePermsConf?.relation ?? PermissionsRelation.and,
                    permissionTreeAttributes: treePermsConf?.permissionTreeAttributes.map(a => a.id) ?? []
                }}
                onSubmitSettings={_handleSubmit}
            />
            <Tab panes={permissionsPanes} className="grow flex-col height100" />
        </>
    );
}

export default NodePermissionTabContent;
