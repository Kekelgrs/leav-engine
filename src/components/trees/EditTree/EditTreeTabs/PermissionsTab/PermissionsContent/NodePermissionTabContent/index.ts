import NodePermissionTabContent from './NodePermissionTabContent';
export default NodePermissionTabContent;
