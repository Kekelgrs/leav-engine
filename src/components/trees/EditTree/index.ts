import EditTree from './EditTree';
export default EditTree;
