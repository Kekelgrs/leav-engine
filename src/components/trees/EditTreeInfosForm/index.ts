import EditTreeInfosForm from './EditTreeInfosForm';
export default EditTreeInfosForm;
