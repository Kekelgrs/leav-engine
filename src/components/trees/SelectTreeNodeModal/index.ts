import SelectTreeNodeModal from './SelectTreeNodeModal';
export default SelectTreeNodeModal;
