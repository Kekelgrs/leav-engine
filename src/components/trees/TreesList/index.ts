import TreesList from './TreesList';
export default TreesList;
