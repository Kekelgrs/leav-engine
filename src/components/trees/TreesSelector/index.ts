import TreesSelector from './TreesSelector';

export default TreesSelector;
