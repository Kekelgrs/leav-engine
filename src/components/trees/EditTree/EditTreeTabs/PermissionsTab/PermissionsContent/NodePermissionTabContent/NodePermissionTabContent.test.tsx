import {render} from 'enzyme';
import React from 'react';
import {mockTreeWithPermConf} from '../../../../../../../__mocks__/trees';
import NodePermissionTabContent from './NodePermissionTabContent';

jest.mock('../../../../../../permissions/PermissionsAttributesSelector', () => {
    return function PermissionsAttributesSelector() {
        return <div>PermissionsAttributesSelector</div>;
    };
});

jest.mock('../../../../../../permissions/DefineTreePermissionsView', () => {
    return function DefineTreePermissionsView() {
        return <div>DefineTreePermissionsView</div>;
    };
});

jest.mock('../../../../../../../hooks/useLang');

describe('NodePermissionTabContent', () => {
    const onSubmit = jest.fn();

    test('Snapshot test', async () => {
        const comp = render(
            <NodePermissionTabContent
                tree={{...mockTreeWithPermConf}}
                library={{...mockTreeWithPermConf.libraries[0]}}
                onSubmitSettings={onSubmit}
                readonly={false}
            />
        );

        expect(comp).toMatchSnapshot();
    });
});
