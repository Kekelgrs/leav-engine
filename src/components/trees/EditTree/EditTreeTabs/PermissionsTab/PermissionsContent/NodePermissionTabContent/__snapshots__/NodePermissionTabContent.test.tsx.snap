// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodePermissionTabContent Snapshot test 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "data": "PermissionsAttributesSelector",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Object {
      "attribs": Object {
        "class": "grow flex-col height100",
      },
      "children": Array [
        Object {
          "attribs": Object {
            "class": "ui attached tabular menu",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "active item",
              },
              "children": Array [
                Object {
                  "data": "Mon Attribut",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "ui bottom attached segment active tab grow flex-col height100",
            },
            "children": Array [
              Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": "DefineTreePermissionsView",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        Object {
          "attribs": Object {
            "class": "ui bottom attached segment active tab grow flex-col height100",
          },
          "children": Array [
            Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": "DefineTreePermissionsView",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "ui attached tabular menu",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "active item",
                },
                "children": Array [
                  Object {
                    "data": "Mon Attribut",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
      ],
      "name": "div",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": [Circular],
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
      },
    },
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
        Object {
          "attribs": Object {
            "class": "grow flex-col height100",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "ui attached tabular menu",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "active item",
                  },
                  "children": Array [
                    Object {
                      "data": "Mon Attribut",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "ui bottom attached segment active tab grow flex-col height100",
                },
                "children": Array [
                  Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "data": "DefineTreePermissionsView",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {},
                    "x-attribsPrefix": Object {},
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "ui bottom attached segment active tab grow flex-col height100",
              },
              "children": Array [
                Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": "DefineTreePermissionsView",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "ui attached tabular menu",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "active item",
                    },
                    "children": Array [
                      Object {
                        "data": "Mon Attribut",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "a",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": null,
          "prev": [Circular],
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "1": Object {
    "attribs": Object {
      "class": "grow flex-col height100",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "ui attached tabular menu",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "active item",
            },
            "children": Array [
              Object {
                "data": "Mon Attribut",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "ui bottom attached segment active tab grow flex-col height100",
          },
          "children": Array [
            Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": "DefineTreePermissionsView",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "ui bottom attached segment active tab grow flex-col height100",
        },
        "children": Array [
          Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "DefineTreePermissionsView",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "ui attached tabular menu",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "active item",
              },
              "children": Array [
                Object {
                  "data": "Mon Attribut",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "data": "PermissionsAttributesSelector",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
      ],
      "name": "div",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": null,
      "prev": null,
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "root": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "PermissionsAttributesSelector",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": null,
          "prev": null,
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
